code/modules/mob/living/living_organs.dm

```diff
@@ -20,19 +20,31 @@
 /mob/living/proc/has_internal_organs()
 	return LAZYLEN(get_internal_organs()) > 0
 
-//Needed for organ surgery
-/mob/living/proc/add_organ(var/obj/item/organ/O, var/obj/item/organ/external/affected = null, var/in_place = FALSE, var/update_icon = TRUE)
-	if(!in_place)
+//Can be called when we want to add an organ in a detached state or an attached state. 
+/mob/living/proc/add_organ(var/obj/item/organ/O, var/obj/item/organ/external/affected = null, var/in_place = FALSE, var/update_icon = TRUE, var/detached = FALSE)
+	. = O.do_install(src, affected, in_place, update_icon, detached)
+	//Only run install effects if we're not detached and we're not adding in place
+	if(!in_place && !(O.status & ORGAN_CUT_AWAY)) 
 		on_gained_organ(O)
-	. = O.do_install(src, affected, in_place, update_icon)
+		updatehealth()
+	return TRUE
 
-/mob/living/proc/remove_organ(var/obj/item/organ/O, var/drop_organ = TRUE, var/detach = TRUE, var/ignore_children = FALSE, var/in_place = FALSE, var/update_icon = TRUE)
-	if(!in_place)
+//Can be called when the organ is detached or attached. 
+/mob/living/proc/remove_organ(var/obj/item/organ/O, var/drop_organ = TRUE, var/detach = FALSE, var/ignore_children = FALSE, var/in_place = FALSE, var/update_icon = TRUE)
+	//Only run effects if we're not already detached, and we're not doing a in-place removal
+	if(!in_place && !(O.status & ORGAN_CUT_AWAY)) //Gotta check the flag here, because of prosthetics handling detached state differently
 		on_lost_organ(O)
 	. = O.do_uninstall(in_place, detach, ignore_children, update_icon)
 
+	if(!in_place && !(O.status & ORGAN_CUT_AWAY)) //Gotta check the flag here, because of prosthetics handling detached state differently
```

Comment:
Is this check duplicated?

Reply:
It is because its the same condition but it needs to happen after do_uninstall, and its too simple of a check for me to bother to place the result into a var.

Reply:
Changed the second condition.